clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusterTierActiveEntity.java

```diff
@@ -559,6 +568,9 @@ public void synchronizeKeyToPassive(PassiveSynchronizationChannel<EhcacheEntityM
     LOGGER.info("Sync started for concurrency key {}.", concurrencyKey);
     if (concurrencyKey == DEFAULT_KEY) {
       stateService.getStateRepositoryManager().syncMessageFor(storeIdentifier).forEach(syncChannel::synchronizeToPassive);
+      Map<Long, Map<Long, EhcacheEntityResponse>> clientSourceIdTrackingMap = messageHandler.getTrackedClients()
```

Comment:
Why not passing the segment, client and messageHandler to the message constructor to let it deal with all this code? I feel it will be easier to test and to read

Reply:
Was actually missing the empty map check which means we do not have to create and send this message.